join.go

```diff
@@ -570,7 +571,7 @@ func (js *joinset) JoinIntoPoint() (models.Point, bool) {
 }
 
 // join all batches the set into a single batch
-func (js *joinset) JoinIntoBatch() (models.Batch, bool) {
+func (js *joinset) JoinIntoBatch(j *JoinNode) (models.Batch, bool) {
```

Comment:
I'd rather the JoinNode get passed into the `newJoinset` function and become a field on the joinset type instead of passing it into each function of the joinset type.